app/assets/javascripts/thredded.es6

```diff
@@ -1,18 +1,59 @@
+//= require babel/polyfill
 //= require jquery
 //= require jquery_ujs
 //= require_tree ./thredded/vendor
 //= require_tree ./thredded
 
-class Thredded {
-  constructor() {
-    this.topicForm = new ThreddedTopicForm();
-    this.privateTopicForm = new ThreddedPrivateTopicForm();
-    this.postForm = new ThreddedPostForm();
-    this.kbShortcuts = new ThreddedKeyboardShortcuts();
-    this.timestamps = new ThreddedTimeStamps();
-    this.searchBar = new ThreddedSearchBar();
-    this.currentlyOnline = new ThreddedCurrentlyOnline();
+(function() {
+
+  const pageComponents = [
+    ['thredded-edit-post', 'postForm'],
+    ['thredded-new-private-topic', 'postForm', 'privateTopicForm'],
+    ['thredded-new-topic', 'postForm', 'topicForm'],
+    ['thredded-posts', 'postForm'],
+    ['thredded-private-topics-index', 'privateTopicForm'],
+    ['thredded-theme', 'topicForm', 'privateTopicForm', 'postForm', 'searchBar'],
+    ['thredded-topics-index', 'topicForm']
+  ];
```

Comment:
What I would like even more is if the components are independent of the pages they are on, so the post form has a `data-thredded-post-form` attribute etc, and the initialization is performed by the component itself (`jQuery($) { $('[data-thredded-post-form]').each(node => new ThreddedPostForm(node)) }`). Is it OK if we do this after this PR is merged?